packages/flutter/lib/src/widgets/splash.dart

```diff
@@ -0,0 +1,623 @@
+///   offering more granular control over UI properties.
+///   (This is especially helpful when its properties come from values that
+///   are already being animated, such as when `Theme.of(context).colorScheme`
+///   inherits from a `MaterialApp`'s `AnimatedTheme`.)
+/// * If a Flutter app isn't using the [Material design system](https://m3.material.io/),
+///   `SplashBox` is the easiest way to add [Splash]es.
+/// {@endtemplate}
+///
+/// Splashes are shown using a [SplashController], a render object that
+/// re-paints each [Splash] when the widget's layout changes.
+///
+/// {@tool snippet}
+/// Generally, a [SplashBox] should be set as the child of widgets that
+/// perform clipping and decoration, and it should be the parent of widgets
+/// that create [Splash]es.
+///
+/// Example:
+///
+/// ```dart
+/// ClipRRect(
+///   borderRadius: BorderRadius.circular(8),
+///   child: ColoredBox(
+///     color: color,
+///     child: const SplashBox(
+///       // add an InkWell here,
+///       // or a different child that creates Splash effects
+///     ),
+///   ),
+/// );
+/// ```
+/// {@end-tool}
+///
+/// {@tool dartpad}
+/// This example shows how to make a button using a [SplashBox].
+///
+/// ** See code in examples/api/lib/widgets/splash_box/splash_box.0.dart **
+/// {@end-tool}
+///
+/// See also:
+/// * [SplashController], used by this widget to enable splash effects.
+/// * [Splash], the class that holds splash effect data.
+class SplashBox extends StatefulWidget {
+  /// {@macro flutter.widgets.splash.SplashBox}
+  const SplashBox({super.key, this.color, this.child});
+
+  /// The value assigned to [SplashController.color].
+  ///
+  /// The [SplashBox] widget doesn't paint this color, but the [child]
+  /// and its descendants can access its value using [Splash.of].
+  ///
+  /// If non-null, the widget will absorb hit tests.
+  final Color? color;
+
+  /// The widget below this widget in the tree.
+  ///
+  /// {@macro flutter.widgets.ProxyWidget.child}
+  final Widget? child;
+
+  @override
+  State<SplashBox> createState() => _SplashBoxState();
+}
+
+class _SplashBoxState extends State<SplashBox> with TickerProviderStateMixin {
+  final GlobalKey _inkFeatureRenderer = GlobalKey(debugLabel: 'ink renderer');
+
+  @override
+  Widget build(BuildContext context) {
+    final Color? color = widget.color;
+    return NotificationListener<LayoutChangedNotification>(
+      onNotification: (LayoutChangedNotification notification) {
+        final SplashController controller = _inkFeatureRenderer.currentContext!.findRenderObject()! as SplashController;
+        controller.didChangeLayout();
+        return false;
+      },
+      child: _InkFeatures(
+        key: _inkFeatureRenderer,
+        color: color,
+        vsync: this,
+        child: widget.child,
+      ),
+    );
+  }
+}
+
+// TODO(nate-thegrate): rename private classes (will involve many test updates)
```

Comment:
> _InkFeatures is a private class that was originally in the same library as the Material widget.
Eventually, I'll probably change the private class names to something without "ink", but surprisingly, doing this would break many different tests, since they're set up to detect the private class name using runtimeType.toString().

Responding to this comment and this todo. Just highlighting that this will be a pain, but probably should be done as a part of the first PR. It's mostly a breaking change on only us, and if this tries to be separated into it's own PR, there's no guarantee that gets landed quickly after. Which leaves some confusing code in the in-between. Ran into the same problem with `toString()` tests with the MaterialState -> WidgetState migration 🙃

Reply:
On the one hand, I'd like to avoid doubling the size of this PR if possible; at the same time, it doesn't feel great to leave it unfinished.

I remember that the submission/approval process for the name changes in #151376 was pretty smooth. Is this something we could try again, or should I just do all the test updates now?

Reply:
> On the one hand, I'd like to avoid doubling the size of this PR if possible; at the same time, it doesn't feel great to leave it unfinished.

I'd probably wait until there's a LGTM on everything else before doing that, for PR readability sake.

> Is this something we could try again, or should I just do all the test updates now?

That was a little different, because it was references to deprecated members in documentation, which is normally fine temporarily. Leaving in private method names of a renamed class can create problems because there's less of a bread crumb trail on why those methods are named that.

Reply:
Sounds good! I'll push a bunch of test updates if/when it would otherwise be ready to merge 👍